modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/batch/ShredJob.scala

```diff
@@ -209,11 +210,18 @@ object ShredJob {
       config
     }
 
-    unshredded.foreach { folder =>
-      System.out.println(s"RDB Shredder: processing $folder")
-      val job = new ShredJob(spark, igluConfig, formats, shredderConfig)
-      val completed = job.run(folder.folderName, atomicLengths, eventsManifest)
-      Discovery.seal(completed, region, sqsQueue)
+    Either.catchNonFatal{
+      unshredded.foreach { folder =>
+        System.out.println(s"RDB Shredder: processing $folder")
+        val job = new ShredJob(spark, igluConfig, config.formats, shredderConfig)
+        val completed = job.run(folder.folderName, atomicLengths, eventsManifest)
+        Discovery.seal(completed, config.region, config.messageQueue)
+      }
+    }.leftMap{ throwable =>
+      config.monitoring.sentry.foreach{ sentry =>
+        System.setProperty("sentry.dsn", sentry.dsn.toString)
```

Comment:
It is unusual to see system properties getting set deep inside the business-logic of a code base.  It is more common to set them early and higher up, during configuration of the app.

Actually my preference is not to set system properties at all, but rather to use `Sentry.init` like how it is done [in the loader](https://github.com/snowplow/snowplow-rdb-loader/blob/master/modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/Environment.scala#L91)